src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -60,19 +70,60 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 _pendingProgress = new PendingProgress();
             }
 
-            _pendingProgress.Update(sourceId, record);
-
             if (_progPane == null)
             {
-                // This is the first time we've received a progress record. Create a progress pane, then show it.
+                // This is the first time we've received a progress record
+                // Create a progress pane
+                // Set up a update flag
+                // Create a timer for updating the flag
 
                 _progPane = new ProgressPane(this);
+
+                if (_progPaneUpdateTimer == null && _progPane != null)
+                {
+                    // Show a progress pane at the first time we've received a progress record
+                    progPaneUpdateFlag = true;
+
+                    // Invoke the timer only once to exclude overlaps
+                    // The timer will be restarted in 'ProgressPaneUpdateTimerElapsed'
+                    _progPaneUpdateTimer = new Timer( new TimerCallback(ProgressPaneUpdateTimerElapsed), null, UpdateTimerThreshold, Timeout.Infinite);
+                }
+            }
+
+            if (progPaneUpdateFlag)
+            {
+                // Update the progress pane only when the timer set up the flag
+                // as a result, we do not block WriteProgress and whole script
+                // and eliminate unnecessary console locks and updates
+                lock (_instanceLock)
+                {
+                    _pendingProgress?.Update(sourceId, record);
+                    _progPane?.Show(_pendingProgress);
+                    progPaneUpdateFlag = false;
```

Comment:
My thoughts were that if we do `UpdateTimerThreshold` large enough the showing can become uneven.
Your offer will be only if machine is too slow. I suppose in this case the PR optimization is generally meaningless.